src/Umbraco.Web.BackOffice/Controllers/TinyMceController.cs

```diff
@@ -65,13 +68,36 @@ public async Task<IActionResult> UploadImage(List<IFormFile> file)
             return new UmbracoProblemResult("Only one file can be uploaded at a time", HttpStatusCode.BadRequest);
         }
 
+        // Create an unique folder path to help with concurrent users to avoid filename clash
+        var imageTempPath =
+            _hostingEnvironment.MapPathContentRoot(Constants.SystemDirectories.TempImageUploads + "/" + Guid.NewGuid());
+
+        // Ensure image temp path exists
+        if (Directory.Exists(imageTempPath) == false)
+        {
+            Directory.CreateDirectory(imageTempPath);
+        }
+
         IFormFile formFile = file.First();
 
         // Really we should only have one file per request to this endpoint
         //  var file = result.FileData[0];
-        var fileName = formFile.FileName.Trim(new[] { '\"' }).TrimEnd();
+        var fileName = formFile.FileName.Trim(new[] {'\"'}).TrimEnd();
         var safeFileName = fileName.ToSafeFileName(_shortStringHelper);
-        var ext = safeFileName.Substring(safeFileName.LastIndexOf('.') + 1).ToLowerInvariant();
+        string ext;
+        var fileExtensionIndex = safeFileName.LastIndexOf('.');
+        if (fileExtensionIndex is not -1)
+        {
+            ext = safeFileName.Substring(fileExtensionIndex + 1).ToLowerInvariant();
```

Comment:
I guess the +1 here is to get the extension without a leading dot.  I think we should also remove the dots from the directionary,. Seems to work with a dot in IsSupportedImageFormat but IsFileAllowedForUpload is not handling the leading dots